src/neo/Persistence/Helper.cs

```diff
@@ -8,5 +9,13 @@ public static byte[] EnsureNotNull(this byte[] source)
         {
             return source ?? Array.Empty<byte>();
         }
+
+        public static void UpdateLocalStateRoot(this SnapshotView snapshot)
+        {
+            snapshot.Storages.Commit();
+            var root = snapshot.LocalStateRoot.GetAndChange(snapshot.Height, () => new HashIndexState());
+            root.Index = snapshot.Height;
+            root.Hash = ((MPTDataCache<StorageKey, StorageItem>)snapshot.Storages).Root.Hash;
```

Comment:
If we require a `MPTDataCache` maybe we should modify https://github.com/neo-project/neo/pull/1692/files#diff-de4093105173aee33135b364a5a08edaR17 to be a `MPTDataCache`

Reply:
But we can't convert `DataClone` to `MPTCache` after `CreateSnapshot` method.
https://github.com/neo-project/neo/blob/835fdeeca87d909fce9b405160102d3250d1bed0/src/neo/IO/Caching/DataCache.cs#L104-L107